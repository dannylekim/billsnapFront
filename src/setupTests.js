// setup file
/** 
 * DON'T TOUCH THIS FILE.
 * IT'S REQUIRED TO TEST COMPONENTS
*/
import Enzyme, { shallow, render, mount } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer  from 'react-test-renderer';

configure({ adapter: new Adapter() });

const matches = children => expect(
    renderer.create(children).toJSON()
  ).toMatchSnapshot();
  
global.shallow = shallow;
global.render = render;
global.mount = mount;
global.matches = matches;
global.localStorage = {
  setItem: jest.fn()
};