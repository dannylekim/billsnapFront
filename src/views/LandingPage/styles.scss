div.page__landing {
  display: -webkit-box;  /* iOS 6-, Safari 3.1-6 */
  display: -ms-flexbox;  /* IE 10 */
  display: -webkit-flex;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 80%;
  height:100%;
  margin-top: 100px;

}

div#flex__item{
  display: flex;
}

div.title__logo{
  font-size: 15px;
}

div.form__inputs{
  width:70%;
  margin-top: 20px;
  margin-left: auto;
  margin-right: auto;
}

input.register__login__inputs{
  font-size: 15x;
  border: 3px solid #47E5B6;
  border-radius: 50px;
}

input.register__login__inputs:hover {
  border: 3px solid #C0FFEC;
}

input.register__login__inputs:active, input.register__login__inputs:focus, input.register__login__inputs:focus:hover {
  border: 3px solid #3BD0A3;
}

button.login_register__submit__button.btn.btn-primary{
  background: #30E3AE;
  border: 1px solid #47E5B6;
  font-size: 15px;
}

button.login_register__submit__button.btn.btn-primary:hover, button.login_register__submit__button.btn.btn-primary:active,
button.login_register__submit__button.btn.btn-primary:focus:hover,button.login_register__submit__button.btn.btn-primary:active:focus{
  background: #30E3AE;
  border: 1px solid #47E5B6;
}

button.form__toggle.btn.btn-primary, button.form__toggle.btn.btn-primary:hover, button.form__toggle.btn.btn-primary:active:focus {
  background-color: #3BD0A3;
  border-color: #3BD0A3;
  margin-left: 15px;
  box-sizing: border-box;
  border-radius: 50px;
}

div.tooltip__inner{
  text-shadow: 1px 1px black;
  border: 1px solid black;
  background-color: rgb(71, 70, 70);
}

div.hidden__div{
  padding: .75rem 1.25rem;
  margin-bottom: 1rem;
  height:48px;
}

h6 a.form__toggle{
  color:rgba(165, 199, 245, 0.801);
}

div.form__seperator{
  display: flex;
  font-size: 18px;
  text-align: center;
}

hr.form__horizontal__line{
  width: 25%;
  border: 0.1px solid rgba(0, 0, 0, 0.63);
}

img.character__icon__image{
  height:50px;
}