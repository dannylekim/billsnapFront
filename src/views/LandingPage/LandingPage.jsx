import React, { useState } from "react";

import { Button } from "shards-react";
import TitleLogo from "../../components/TitleLogo";
import RegisterForm from "../../components/RegisterForm";
import LoginForm from "../../components/LoginForm";
import "./styles.scss";

/**
 * @description container component that should be in the middle of the login page
 */

export default props => {
  const [formType, setFormType] = useState("register");

  return (
    <div className="page__register">
      <div className="title-logo">
        <TitleLogo />
      </div>
      <div className="register__container">
        {formType === "register" ? <RegisterForm /> : <LoginForm />}
        <div>
          <h6>
            {formType === "login" ? "New to Billsnap? " : "Have an account? "}
            <Button
              className="login-link"
              onClick={() =>
                setFormType(formType === "login" ? "register" : "login")
              }
            >
              {formType === "login"
                ? "Register an account."
                : "Login to your account."}{" "}
            </Button>
          </h6>
        </div>
      </div>
    </div>
  );
};
