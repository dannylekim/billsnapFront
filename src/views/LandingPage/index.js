import LandingPage from './LandingPage.jsx';
import { connect } from 'react-redux'
import { toggleLandingPageForm } from '../../redux/actions/applicationActions';

const mapStateToProps = (state) => ({
    formType: state.application.showRegisterFirst ? 'register' : 'login'
  })
  
const mapDispatchToProps = (dispatch) => ({
    toggleFormType: () => dispatch(toggleLandingPageForm())
})

export default connect(mapStateToProps, mapDispatchToProps)(LandingPage);