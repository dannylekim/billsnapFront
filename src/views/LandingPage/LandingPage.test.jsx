import React from "react";
import LandingPage from "./LandingPage.jsx";
import {matches} from "../../setupTests";

describe("LandingPage", () => {


  describe("render", () => {
    describe("snapshots ðŸ“¸", () => {
      it("LandingPage should match snap shot", () => {
        matches(<LandingPage/>);
      });
    });

  });
});

