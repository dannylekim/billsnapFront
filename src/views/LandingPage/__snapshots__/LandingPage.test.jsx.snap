// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage render snapshots ðŸ“¸ LandingPage should match snap shot 1`] = `
<div
  className="page__register"
>
  <div
    className="title-logo"
  >
    <div
      className="title"
    >
      <div
        className="title__row"
      >
        <div
          className="title__title"
        >
          <h1>
            Billsnap
          </h1>
        </div>
        <div
          className="title__logo"
        >
          <img
            alt="Logo"
            src="https://via.placeholder.com/150x100.png?text=150x100+Logo"
          />
        </div>
      </div>
      <div
        className="title__row title__text"
      >
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Atque dignissimos eos error quo, repellendus repudiandae sunt. Amet asperiores atque commodi, deleniti dicta expedita ipsa modi nemo, quos rem voluptate voluptates.
        </p>
      </div>
    </div>
  </div>
  <div
    className="register__container"
  >
    <div
      className="regForm"
    >
      <div
        className="register__form"
      >
        <form
          className=""
        >
          <div
            className="form-inputs"
          >
            <div
              className="form-group"
              onChange={[Function]}
            >
              <input
                className="mb-2 form-control"
                id="firstName"
                name="firstName"
                placeholder="Enter your first name"
                type="text"
              />
            </div>
            <div
              className="form-group"
              onChange={[Function]}
            >
              <input
                className="mb-2 form-control"
                id="middleName"
                name="middleName"
                placeholder="Enter your middle name"
                type="text"
              />
            </div>
            <div
              className="form-group"
              onChange={[Function]}
            >
              <input
                className="mb-2 form-control"
                id="lastName"
                name="lastName"
                placeholder="Enter your last name"
                type="text"
              />
            </div>
            <div
              className="form-group"
              onChange={[Function]}
            >
              <input
                className="mb-2 form-control"
                id="birthDate"
                name="birthDate"
                placeholder="Enter your date of birth"
                type="date"
              />
            </div>
            <div
              className="form-group"
              onChange={[Function]}
            >
              <input
                className="mb-2 form-control"
                id="email"
                name="email"
                placeholder="Enter your email"
                type="email"
              />
            </div>
            <div
              className="form-group"
              onChange={[Function]}
            >
              <input
                className="mb-2 form-control"
                id="phoneNumber"
                name="phoneNumber"
                placeholder="Enter your phone number"
                type="tel"
              />
            </div>
            <div
              className="form-group"
              onChange={[Function]}
            >
              <input
                className="mb-2 form-control"
                id="password"
                name="password"
                placeholder="Enter a password"
                type="password"
              />
            </div>
            <div
              className="form-group"
              onChange={[Function]}
            >
              <input
                className="mb-2 form-control"
                id="confirmPassword"
                name="confirmPassword"
                placeholder="Verify password"
                type="password"
              />
            </div>
            <div
              className="form-group"
              onChange={[Function]}
            >
              <input
                className="mb-2 form-control"
                id="location"
                name="location"
                placeholder="Enter your location"
                type="text"
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <button
              className="btn btn-dark btn-lg btn-pill"
              name="submit"
              onClick={[Function]}
              type="button"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
    <div>
      <h6>
        Have an account? 
        <button
          className="login-link btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          Login to your account.
           
        </button>
      </h6>
    </div>
  </div>
</div>
`;
